### PR TITLE
Revamp records related BBEs

### PR DESCRIPTION
**Pull request by lasinicl**
## Purpose
Revamps the following BBEs

- Open records
- Controlling openness
- Type inclusion for records
- Default values for record fields

Fixes https://github.com/ballerina-platform/ballerina-distribution/issues/3709

### CODE REVIEW

File: examples/default-values-for-record-fields/default_values_for_record_fields.bal [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
-    io:println(check j.cloneWithType(Employee));
+    // Creates a `Student` record without explicitly specifying a value for the `code` field.
+    Student s1 = {name: "Anne", age: 23};
+    io:println(s1);
 
-    // `*Employee` copies the default values.
-    ContractEmployee emp = {name: "Anne", department: "HR", duration: 12};
-    io:println(emp);
+    json j = {name: "Anne", age: 23};
+    // Calling the `value:cloneWithType()` function with `Student` will make use of default values
```
Comment (praneesha) [inline examples/default-values-for-record-fields/default_values_for_record_fields.bal:21]:
```suggestion
    // Calling the `value:cloneWithType()` function with `Student` will make use of the default values
```

File: examples/default-values-for-record-fields/default_values_for_record_fields.md [hunk 2]
```diff
@@ -1,7 +1,12 @@
 # Default values for record fields
 
-Ballerina allows default values for record fields as part of the record's type descriptor. A default value is an expression. Default values do not affect static typing. They only affect the use of type descriptors to construct records. Calling the `value:cloneWithType()` function with a record type-descriptor `T` will make use of default values in `T` if required. Similarly, using `*T` also copies the default values.
+Ballerina allows default values for record fields as part of the record's type descriptor. A default value is an expression. Default values do not affect static typing. They only affect the use of type descriptors to construct records. Calling the `value:cloneWithType()` function with a record type descriptor `T` will make use of default values in `T` if required. Similarly, using `*T` in type inclusions also copies the default values.
```
Comment (praneesha) [inline examples/default-values-for-record-fields/default_values_for_record_fields.md:3]:
```suggestion
Ballerina allows default values for record fields as part of the record's type descriptor. A default value is an expression. Default values do not affect static typing. They only affect the use of type descriptors to construct records. Calling the `value:cloneWithType()` function with a record type descriptor `T` will make use of the default values in `T` if required. Similarly, using `*T` in type inclusions also copies the default values.
```

File: examples/default-values-for-record-fields/default_values_for_record_fields.metatags [hunk 3]
```diff
@@ -1,2 +1,2 @@
-description: This BBE demonstrates default values for record fields in Ballerina.
-keywords: ballerina, ballerina by example, bbe, default value, record, record field
+description: This BBE demonstrates default values for record fields, adding default values to record fields, default values in inclusions.
```
Comment (praneesha) [inline examples/default-values-for-record-fields/default_values_for_record_fields.metatags:1]:
```suggestion
description: This BBE demonstrates the default values for record fields, adding default values to record fields, and default values in inclusions.
```

File: examples/open-records/open_records.bal [hunk 4]
```diff
@@ -1,32 +1,56 @@
...
 
-// `Employee` type allows additional fields with `anydata` values.
-type Employee record {
+type PartTimeStudent record {|
     string name;
-    int id;
-};
+    int age;
+    // Rest descriptor allows additional fields with `anydata` values
+    // in `PartTimeStudent` type.
```
Comment (praneesha) [inline examples/open-records/open_records.bal:13]:
```suggestion
    // in the `PartTimeStudent` type.
```

File: examples/type-inclusion-for-records/type_inclusion_for_records.bal [hunk 5]
```diff
@@ -1,33 +1,29 @@
...
+|};
 
-// `Time` record has all the fields of `Date` and `TimeOfDay`.
-type Time record {
-    *Date;
-    *TimeOfDay;
-    // Overrides `hour` field in `TimeOfDay`.
-    int:Unsigned8 hour;
-};
+// `PartTimeStudent` record has all the fields of `Person` and `Student`.
```
Comment (praneesha) [inline examples/type-inclusion-for-records/type_inclusion_for_records.bal:13]:
```suggestion
// The `PartTimeStudent` record has all the fields of `Person` and `Student`.
```

File: examples/type-inclusion-for-records/type_inclusion_for_records.bal [hunk 6]
```diff
@@ -1,33 +1,29 @@
...
+type PartTimeStudent record {|
+    *Person;
+    *Student;
+    // Overrides `code` field in `Student`.
```
Comment (praneesha) [inline examples/type-inclusion-for-records/type_inclusion_for_records.bal:17]:
```suggestion
    // Overrides the `code` field in `Student`.
```

File: examples/type-inclusion-for-records/type_inclusion_for_records.md [hunk 7]
```diff
@@ -1,7 +1,12 @@
 # Type inclusion for records
 
-Type inclusion enables you to create a record by combining fields of other records. Using the `*T` notation, you can include the record type T in the record type descriptor of another record. This is effectively the same as copying the fields of the included records into the including record.
+Type inclusion enables you to create a record by combining fields of other records. Using the `*T` notation, you can include the record type `T` in the record type descriptor of another record. This is effectively the same as copying the fields of the included records into the including record.
```
Comment (praneesha) [inline examples/type-inclusion-for-records/type_inclusion_for_records.md:3]:
```suggestion
Type inclusion enables you to create a record by combining fields of other records. You can include the record type `T` in the record type descriptor of another record by using the `*T` notation. This is effectively the same as copying the fields of the included records into the including record.
```

File: examples/type-inclusion-for-records/type_inclusion_for_records.metatags [hunk 8]
```diff
@@ -1,2 +1,2 @@
-description: This BBE demonstrates type inclusion for records in Ballerina.
-keywords: ballerina, ballerina by example, bbe, record, type inclusion
+description: This BBE demonstrates type inclusion for records, record type inclusion, inclusing fields of a record.
```
Comment (praneesha) [inline examples/type-inclusion-for-records/type_inclusion_for_records.metatags:1]:
```suggestion
description: This BBE demonstrates type inclusion for records, record type inclusion, and including fields of a record.
```
Check if the technical meaning is affected by the changes.

File: examples/controlling-openness/controlling_openness.bal [hunk 9]
```diff
@@ -1,29 +1,37 @@
...
-
-// `x` is a `map` with `float` values.
-map<float> m1 = x;
-
-type Headers record {|
-    string 'from;
-    string to;
+type PartTimeStudent record {|
+    string name;
+    string country;
```
Comment (prakanth97) [inline examples/controlling-openness/controlling_openness.bal:10]: Shall we use inclusion here?
```suggestion
    *Student;
```

File: examples/default-values-for-record-fields/default_values_for_record_fields.bal [hunk 10]
```diff
@@ -1,23 +1,29 @@
...
-    // `*Employee` copies the default values.
-    ContractEmployee emp = {name: "Anne", department: "HR", duration: 12};
-    io:println(emp);
+    json j = {name: "Anne", age: 23};
+    // Calling the `value:cloneWithType()` function with `Student` will make use of the default values
+    // in `Student`.
+    Student s2 = check j.cloneWithType(Student);
+    io:println(s2);
+
+    // `*Student` copies the default values.
```
Comment (prakanth97) [inline examples/default-values-for-record-fields/default_values_for_record_fields.bal:26]:
```suggestion
    // `*Student` includes the default values.
```

Comment (KavinduZoysa) [reply examples/default-values-for-record-fields/default_values_for_record_fields.bal:26]: I think the word `copy` is more suitable for this place. If we consider `*` as an operation, `*Student` is copying fields with default values.

**Review by praneesha — APPROVED** [review]

**Pull request merged**